src/sass/checkboxes/_base.scss

```diff
@@ -3,121 +3,107 @@
-     */
-    margin-left: .2rem;
-    font-weight: $font-weight-regular;
-
-    &::before {
-      display: inline-block;
-      content: ' ';
-      border-radius: .125rem;
-      margin-right: .75rem;
-      width: $spacing-sm;
-      height: $spacing-sm;
-      transition: all .2s ease;
-
-      /**
-       * There are some magic numbers here that help improve rendering
-       * of box-shadows in Firefox and IE. Making them just a smidge
-       * wider (.08rem instead of .0625rem/1px) than one pixel compensates
-       * for some weird aliasing that is happening. As a bonus it seem to
-       *  make radios and checkboxes look optically closer the 1px border
-       * used on other inputs like text inputs, textareas, and selects.
-       */
-      box-shadow: 0 0 0 .08rem $color-black-400;
-
-      /**
-       * This helps visually center the label with the pseudo
-       * element checkbox.
-       */
-      vertical-align: middle;
-      margin-top: -.1rem;
-    }
-
-    &:hover {
-      cursor: pointer;
-    }
+.rvt-checkbox {
+  display: inline-block;
+  padding-left: $spacing-lg;
+  position: relative;
+
+  input[type=checkbox] {
+    cursor: pointer;
+    height: $spacing-md;
+    left: 0;
+    margin: 0;
+    top: 0;
+    width: $spacing-lg;
+    opacity: 0;
+    position: absolute;
   }
 
-  &:checked + .#{$prefix}-label,
-  .#{$prefix}-checkbox-wrapper &:checked ~ .#{$prefix}-label {
-    &::before {
-      color: $color-white-base;
-      background-color: $color-blue-600;
-      box-shadow: 0 0 0 .08rem $color-blue-600;
-      background-image: url('data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHdpZHRoPSIxNiIgaGVpZ2h0PSIxNiIgdmlld0JveD0iMCAwIDE2IDE2Ij4KICAgIDxwYXRoIGZpbGw9IiNmZmZmZmYiIGQ9Ik02LjcyLDEzbC0uNDgtLjM2LTMtM0ExLDEsMCwwLDEsNC43MSw4LjI5bDIuMTEsMi4xMiw0LjMzLTYuOTRhMSwxLDAsMCwxLDEuNywxLjA2TDcuNjQsMTIuODdaIi8+Cjwvc3ZnPgo=');
-      background-position: 50% 50%;
-      background-repeat: no-repeat;
-    }
+  &--sr-only-label {
+    padding-left: $spacing-md;
   }
 
-  /* stylelint-disable */
+  &--sr-only-label input[type=checkbox] {
+    width: $spacing-md;
+  }
 
-  &:focus + .#{$prefix}-label::before,
-  .#{$prefix}-checkbox-wrapper &:focus ~ .#{$prefix}-label::before {
-    box-shadow:
-      0 0 0 .08rem $color-black-700,
-      0 0 0 .1875rem $color-white-base,
-      0 0 0 .3125rem $color-blue-600;
+  &--sr-only-label input[type=checkbox] ~ label {
+    height: 0;
+    overflow: hidden;
+    padding: 0;
+    width: 0;
   }
 
-  /* stylelint-enable */
+  input[type=checkbox] ~ label {
+    display: inline-block;
 
-  &:disabled + .#{$prefix}-label::before,
-  .#{$prefix}-checkbox-wrapper &:disabled ~ .#{$prefix}-label::before {
-    background-color: $color-black-200;
-    box-shadow: 0 0 0 .08rem $color-black-300;
+    &:hover {
+      cursor: pointer;
+    }
   }
 
-  /* stylelint-disable */
+  input[type=checkbox] ~ label::before {
+    border-radius: $spacing-xxs / 2;
```

Comment:
While it looks fine in Chrome and Safari, Firefox has some odd rendering issues with rounded corners and box-shadow. Some of the background color bleeds through at the corners.

You could resolve this by replacing the box-shadow that fakes a border with a literal `2px` border. In doing this, it may simplify other parts in this file. But you will need to shift the left and top position by `2px` to put it back in the same visual position as it currently is.

If this is too nit-picky, that's fine. Just trying to solve problems as I see them, no matter how minor.

Reply:
I'm also fine with doing this work in another branch and merging it in, if you don't want to do it. It may give me good practice with contributing to this repo.

Reply:
That's actually a good idea. I'm fine with tackling that as a separate issue later.